packages/lib/src/vis/shared/VisCanvas.tsx

```diff
@@ -1,77 +1,84 @@
 import { useMeasure } from '@react-hookz/web';
 import { Canvas } from '@react-three/fiber';
+import type { PropsWithChildren } from 'react';
 
-import { computeCanvasSize, getAxisOffsets } from '../utils';
+import type { AxisConfig } from '../models';
+import { getSizeToFit, getAxisOffsets } from '../utils';
 import AxisSystem from './AxisSystem';
 import AxisSystemProvider from './AxisSystemProvider';
-import type { AxisSystemProviderProps } from './AxisSystemProvider';
 import styles from './VisCanvas.module.css';
 
-interface Props extends AxisSystemProviderProps {
-  aspectRatio?: number;
+interface Props {
   title?: string;
+  canvasRatio?: number | undefined;
+  visRatio?: number | undefined;
+  abscissaConfig: AxisConfig;
+  ordinateConfig: AxisConfig;
 }
 
-function VisCanvas(props: Props) {
-  const { aspectRatio, title, children, ...providerProps } = props;
-  const { abscissaConfig, ordinateConfig } = providerProps;
+function VisCanvas(props: PropsWithChildren<Props>) {
+  const {
+    title,
+    canvasRatio,
+    visRatio,
+    abscissaConfig,
+    ordinateConfig,
+    children,
+  } = props;
 
-  const [visAreaSize, visAreaRef] = useMeasure<HTMLDivElement>();
-  if (!visAreaSize) {
-    return <div ref={visAreaRef} className={styles.visArea} />;
-  }
+  const shouldMeasure = !!canvasRatio;
+  const [areaSize, areaRef] = useMeasure<HTMLDivElement>();
+  const canvasSize = areaSize && getSizeToFit(areaSize, canvasRatio);
 
   const axisOffsets = getAxisOffsets({
     left: !!ordinateConfig.label,
     bottom: !!abscissaConfig.label,
     top: !!title,
   });
 
-  const availableSize = {
-    width: visAreaSize.width - axisOffsets.left - axisOffsets.right,
-    height: visAreaSize.height - axisOffsets.bottom - axisOffsets.top,
-  };
-
-  const visSize = computeCanvasSize(availableSize, aspectRatio);
-
   return (
-    <div ref={visAreaRef} className={styles.visArea}>
-      <div
-        className={styles.vis}
-        style={{
-          ...visSize,
-          paddingBottom: axisOffsets.bottom,
-          paddingLeft: axisOffsets.left,
-          paddingTop: axisOffsets.top,
-          paddingRight: axisOffsets.right,
-          boxSizing: 'content-box',
```

Comment:
The padding is now on the parent, which means I no longer need to use `boxSizing`.